perfkitbenchmarker/linux_benchmarks/aws_dynamodb_ycsb_benchmark.py

```diff
@@ -0,0 +1,162 @@
+flags.DEFINE_enum('aws_dynamodb_ycsb_consistentReads',
+                  None, ['false', 'true'],
+                  "Consistent reads cost 2x eventual reads. "
+                  "'false' is default which is eventual")
+
+
+def GetConfig(user_config):
+    config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+    if FLAGS['ycsb_client_vms'].present:
+        config['vm_groups']['default']['vm_count'] = FLAGS.ycsb_client_vms
+    return config
+
+
+def CheckPrerequisites(benchmark_config):
+    """Verifies that the required resources are present.
+    Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+    """
+    ycsb.CheckPrerequisites()
+
+
+def Prepare(benchmark_spec):
+    """Install YCSB on the target vm.
+    Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+    """
+    if FLAGS.aws_dynamodb_ycsb_dynamodb_region is None:
+      prereg = FLAGS.zones[0]
+    else:
+      prereg = FLAGS.aws_dynamodb_ycsb_dynamodb_region
+    benchmark_spec.always_call_cleanup = True
+    benchmark_spec.dynamodb_instance = aws_dynamodb.AwsDynamoDBInstance(
+        region=prereg,
+        table_name='{0}-{1}'.format(FLAGS.aws_dynamodb_ycsb_table, FLAGS.run_uri),
+        primary_key=FLAGS.aws_dynamodb_ycsb_dynamodb_primarykey)
+    if benchmark_spec.dynamodb_instance.Exists():
+      logging.warning('DynamoDB table {0} exists, delete it first.'
+                      .format(FLAGS.aws_dynamodb_ycsb_table, FLAGS.run_uri)),
+    benchmark_spec.dynamodb_instance._Delete()
+    benchmark_spec.dynamodb_instance._Create()
+    if not benchmark_spec.dynamodb_instance.Exists():
+      logging.warning('Failed to create DynamoDB table.')
+    benchmark_spec.dynamodb_instance._Delete()
+    vms = benchmark_spec.vms
+    # Install required packages.
+    vm_util.RunThreaded(_Install, vms)
+    benchmark_spec.executor = ycsb.YCSBExecutor('dynamodb')
+
+
+def Run(benchmark_spec):
+    """Run YCSB on the target vm.
+    Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+    required to run the benchmark.
+    Returns:
+    A list of sample.Sample objects.
+    """
+    if FLAGS.aws_dynamodb_ycsb_dynamodb_region is None:
+      runreg = FLAGS.zones[0]
+    else:
+      runreg = FLAGS.aws_dynamodb_ycsb_dynamodb_region
+    vms = benchmark_spec.vms
+
+    run_kwargs = {
+        'dynamodb.awsCredentialsFile': AWS_CREDENTIAL_DIR,
+        'dynamodb.primaryKey': FLAGS.aws_dynamodb_ycsb_dynamodb_primarykey,
+        'dynamodb.endpoint': 'http://dynamodb.{0}.amazonaws.com'
+                             .format(runreg),
+        'readproportion': FLAGS.aws_dynamodb_ycsb_readproportion,
+        'updateproportion': FLAGS.aws_dynamodb_ycsb_updateproportion,
+        'table': '{0}-{1}'.format(FLAGS.aws_dynamodb_ycsb_table, FLAGS.run_uri),
+        'dynamodb.consistentReads': FLAGS.aws_dynamodb_ycsb_consistentReads,
```

Comment:
aws_dynamodb_ycsb_consistentReads does not have default value? does ycsb work with None?

Reply:
ycsb did seem to work with none, but set a default to be more clear.